js/ui/codeView.js

```diff
@@ -435,17 +445,36 @@ var CodingSession = GObject.registerClass({
         this._app = value;
         if (this._app)
             this._setupAppWindow();
+        this.notify('app');
     }
 
     get app() {
         return this._app;
     }
 
+    set state(value) {
+        this._state = value;
+        this.notify('state');
+    }
+
+    get state() {
+        return this._state;
+    }
+
     set toolbox(value) {
         this._cleanupToolboxWindow();
         this._toolbox = value;
         if (this._toolbox)
             this._setupToolboxWindow();
+        this.notify('toolbox');
+    }
+
+    set hackableApp(value) {
```

Comment:
What I had in mind for this was to move the responsibility for creation/state updating of the `HackableApp` instance entirely to `CodingSession`. The getter for `hackableApp()` can create it if one does not exist yet.
I think that would result in a smaller patch, and we may not even need the separate `HackableApp` class. Basically the `HackableApp` interface is how a `CodingSession` exposes itself over D-Bus.

Reply:
I understand is that there is no need for a HackableApp class since its variables like `_dbus_impl` and other stuff could be put directly inside the CodingSession. So this class could just be removed.

If the HackableApp class gets removed. Where should I export the CodingSession/HackableApp object path (i.e. /com/endlessm/HackableApp/1)? If this is in the CodingSession constructor, then I should modify it to receive an object path, or maybe I can add an `export()` method to the CodingSession, so the HackableAppsManager calls `export` whenever a new session is added. 

---
If the HackableApp is not needed because it would be represented by the CodingSession. Wouldn't the same apply for HackableAppsManager, since this is like a dbus representation of the CodeViewManager?